From 0521673f4e76e8f768c6453648170b1e701317af Mon Sep 17 00:00:00 2001
From: Donald Sharp <sharpd@nvidia.com>
Date: Tue, 27 May 2025 16:11:41 -0400
Subject: [PATCH] zebra: Free up tt when failure mode happens

Coverity found a code path where tt is not freed up.

Signed-off-by: Donald Sharp <sharpd@nvidia.com>
---
 zebra/zebra_srv6_vty.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/zebra/zebra_srv6_vty.c b/zebra/zebra_srv6_vty.c
index 80d8120ace6c..f43ffd1705d7 100644
--- a/zebra/zebra_srv6_vty.c
+++ b/zebra/zebra_srv6_vty.c
@@ -496,11 +496,15 @@ static void do_show_srv6_sid_specific(struct vty *vty, json_object **json,
 		ttable_restyle(tt);
 		ttable_rowseps(tt, 0, BOTTOM, true, '-');
 
-		if (!sid_ctx || !sid_ctx->sid)
+		if (!sid_ctx || !sid_ctx->sid) {
+			ttable_del(tt);
 			return;
+		}
 
-		if (locator && sid_ctx->sid->locator != locator)
+		if (locator && sid_ctx->sid->locator != locator) {
+			ttable_del(tt);
 			return;
+		}
 
 		do_show_srv6_sid_line(tt, sid_ctx->sid);
 
